### DISCUSSION TITLE
Does Flet support Hot Reload concept ?

### DISCUSSION THREAD

**Discussion by TheCodeCache** [flet-dev/flet#1024] [Q&A] [answered]
There was a lot of `context switch` while developing a simple UI using `Flet`. 
While Flet app is up & running, any changes to python flet code does not reflect on UI. I need to restart the application every time I do any code changes.

This is not the case with `Flutter`. UI screens are reflecting immediately as I perform code changes in `VS Code` Editor.

I do aware of hot reload feature available in PyCharm "`Professional`" Edition (which is not free), 
however, I'm not sure whether it will work with Flet or not

Please share the thoughts around the same and propose some alternatives to improve the productivity with Flet UI development.

**Comment by TheCodeCache** [comment, marked as answer]
That's awesome : ) 

Thank you @FeodorFitsner for this [`answer`](https://github.com/flet-dev/flet/discussions/241#discussioncomment-3491272)

However, I'm curious to know - does it launch a `webserver` internally, if that is so, which web server is this, is it `Apache` / `Nginx` ? 
and whether `communication` with webserver is `secured` ? please correct me if my thoughts are in different direction

 Thank you for helping all of us with this amazing package

**Reply by FeodorFitsner** [reply]
When you run a Flet program in Desktop mode no web server is started. Instead, on Windows it starts a TCP socket server bound to 127.0.0.1 and random port and on Linux/macOS it starts a unix domain socket server on a random file name: https://github.com/flet-dev/flet/pull/838

**Reply by TheCodeCache** [reply]
Thanks @FeodorFitsner for the clarification.

Just a doubt on hot-reload feature in Flet, does it support only single file reload ?, actually I tried with multi file reload, it seemed not working,

For ex: consider below dummy code structure - 

<img width="175" alt="image" src="https://user-images.githubusercontent.com/26399543/218248219-e674a53d-5066-4d7b-8073-005d3de677da.png">


`main.py` file uses `hello.py` as dependency like below - 

`hello.py` contents - 
```
""" hello.py """
def print():
    return "hello from 'hello' module"
```

`main.py` contents - 
```
""" main.py """
import folder1.folder2.hello
import flet as ft
print(f"message from hello.py : {hello.print()}")
```

When I run `flet main.py`, the flet UI app comes up, and all the changes which I do in `main.py` gets reflected on the flet UI . However, if I do any changes in its dependent module `hello.py`, the same is not propagating to `main.py` and fails to reflect on flet UI app

Kindly advice how to `hot-reload` multiple files across whole project using `Flet` ?

Reference/Note: [`Reloadium`](https://reloadium.io/features/general-python) is a plugin in PyCharm that supports `multi-file reload`

**Reply by TheCodeCache** [reply]
Thanks to [this](https://flet.dev/docs/guides/python/hot-reload), I found a way to perform reload across multiple files using below command - 

```
flet run <your_program.py> -d
``` 

For now, my issue is resolved using above solution, I'll get back in case of any further issues,

Thank you!
